### PR TITLE
Show an alert if Live Activities are disabled

### LINKED ISSUE #216: Show an alert if Live Activities are disabled
We should show an alert with a link to the app's settings if the user disabled Live Activities.

### PR DESCRIPTION
**Pull request by planecore**
Closes #216

### CODE REVIEW

File: app/i18n/he.json [hunk 1]
```diff
@@ -160,7 +160,9 @@
     "departsIn": "הרכבת יוצאת בעוד %{minutes} דק'",
     "departsNow": "הרכבת יוצאת עכשיו",
     "rideExistsTitle": "קיימת נסיעה פעילה",
-    "rideExistsMessage": "עצרו את הנסיעה הנוכחית על מנת להתחיל חדשה"
+    "rideExistsMessage": "עצרו את הנסיעה הנוכחית על מנת להתחיל חדשה",
+    "disabledTitle": "פעילויות בזמן אמת כבוי",
```
Comment (guytepper) [inline app/i18n/he.json:164]:
```suggestion
    "disabledTitle": "פעילויות בזמן אמת כבויות",
```

File: app/i18n/he.json [hunk 2]
```diff
@@ -160,7 +160,9 @@
...
+    "disabledMessage": "אפשרו פעילויות בזמן אמת בהגדרות על מנת להתחיל נסיעה"
```
Comment (guytepper) [inline app/i18n/he.json:165]:
```suggestion
    "disabledMessage": "יש לאפשר פעילויות בזמן אמת בהגדרות המכשיר על מנת להתחיל נסיעה"
```

File: app/i18n/en.json [hunk 3]
```diff
@@ -155,9 +155,11 @@
     "arrived": "You have arrived",
     "departsIn": "Train departs in %{minutes} min",
     "departsNow": "Train departs now",
+    "trainArrives": "Train arrives to destination",
     "rideExistsTitle": "An active ride exists",
     "rideExistsMessage": "Cancel the current ride in order to start a new one",
-    "trainArrives": "Train arrives to destination"
+    "disabledTitle": "Live Activities disabled",
+    "disabledMessage": "Open settings and enable Live Activities to be able to start a ride"
```
Comment (guytepper) [inline app/i18n/en.json:162]:
```suggestion
    "disabledMessage": "To start a ride, please open your device settings and enable Live Activities."
```

File: app/app.tsx [hunk 4]
```diff
@@ -73,9 +73,14 @@ function App() {
     const subscription = AppState.addEventListener("change", (nextAppState) => {
       if (appState.current.match(/inactive|background/) && nextAppState === "active") {
         // App has come to the foreground!
-        // Check if the current ride id is still active
-        if (rootStore && rootStore.ride.id) {
-          rootStore.ride.isRideActive(rootStore.ride.id)
+        if (rootStore) {
+          // Check Live Activities authorization
+          rootStore.ride.checkActivityAuthorizationInfo()
```
Comment (guytepper) [inline app/app.tsx:78]: You are already calling this in the model `afterCreate()` method. Isn't it sufficient?

File: app/models/ride/ride.ts [hunk 5]
```diff
@@ -48,13 +57,23 @@ export const RideModel = types
     },
   }))
   .actions((self) => ({
+    setActivityAuthorizationInfo(newInfo: ActivityAuthorizationInfo) {
+      self.activityAuthorizationInfo = newInfo
+    },
+    async checkActivityAuthorizationInfo() {
+      if (Platform.OS === "ios") {
```
Comment (guytepper) [inline app/models/ride/ride.ts:64]: It's better to have this condition in the calling site

File: app/screens/route-details/components/start-ride-button.tsx [hunk 6]
```diff
@@ -79,6 +80,17 @@ export const StartRideButton = observer(function StartRideButton(props: StartRid
           analytics().logEvent("start_live_ride_disable_press")
```
Comment (guytepper) [inline app/screens/route-details/components/start-ride-button.tsx:80]: `_disabled_press`

It's worth adding the disabled reason as a parameter to the event

File: ios/BetterRail/RNBetterRail.swift [hunk 7]
```diff
@@ -109,4 +110,13 @@ class RNBetterRail: NSObject {
       }
     }
   }
+
+  @available(iOS 16.2, *)
+  @objc func activityAuthorizationInfo(_ blackString: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) -> Void {
```
Comment (guytepper) [inline ios/BetterRail/RNBetterRail.swift:115]: Why is the empty string needed?  


Also:
```suggestion
  @objc func activityAuthorizationInfo(_ emptyString: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) -> Void {
```

**Review by guytepper — CHANGES_REQUESTED** [review]
👑

File: ios/BetterRail/RNBetterRail.swift (hunk 7 above)
Comment (planecore) [reply ios/BetterRail/RNBetterRail.swift:115]: `resolve` must be the second parameter or later, so we must do this hack

File: app/app.tsx (hunk 4 above)
Comment (planecore) [reply app/app.tsx:78]: I want to check it every time the user opens the app, to update the state when the user comes back from settings

**Review by guytepper — APPROVED** [review]

**Pull request merged**